Examples/restcomm-olympus-nopod/restcomm-olympus/MainTableViewController.mm

```diff
@@ -187,6 +187,55 @@ - (void)viewDidLoad {
     self.spinner = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleGray];
     //get contacts first time
     [self checkContactsAccess];
+    
+#warning this will be moved into app delegate; for now we are handling it here
+    //if there is a token we got it before openning this viewcontroller (this is only
+    //for the first time when user takes up time to login
+    NSUserDefaults* appDefaults = [NSUserDefaults standardUserDefaults];
+    NSString *deviceToken = [appDefaults objectForKey:@"deviceToken"];
+    
+    [self registerForPush:deviceToken];
+    
+    if (!deviceToken){
+        //otherwise subscribe to notification
+        [[NSNotificationCenter defaultCenter] addObserver:self
+                                                 selector:@selector(tokenReceivedNotification:)
+                                                     name:@"TokenReceivedNotification"
+                                                   object:nil];
+        
+    }
+    
+}
+
+- (void) tokenReceivedNotification:(NSNotification *) notification
+{
+    if ([[notification name] isEqualToString:@"TokenReceivedNotification"]){
+        NSDictionary *userInfo = notification.userInfo;
+        NSString *deviceToken = [userInfo objectForKey:@"deviceTokenKey"];
+        [self registerForPush:deviceToken];
+    }
+}
+
+- (void) registerForPush:(NSString *)token{
+   if (token){
+        NSString *pushCertificatesPathPublic = [[NSBundle mainBundle] pathForResource:@"certificate_key_push" ofType:@"pem"];
+        NSString *pushCertificatesPathPrivate = [[NSBundle mainBundle] pathForResource:@"rsa_private_key_push" ofType:@"pem"];
+        NSMutableDictionary *dic = [[NSMutableDictionary alloc] initWithObjectsAndKeys:
```

Comment:
I think we need to start getting into the habit of adding proper constants for key names for API related dictionaries. Can you provide such things for keys like 'friendly-name', 'username', etc?

Reply:
ok